### ISSUE TITLE
Unable to upload Shiny app

### ISSUE THREAD

**Issue by philstraforelli** [STAT545-UBC/Discussion#437]
Hello, 

I'm getting the following error when I attempt to upload my app to shinyapps.io:

>Error: Unhandled Exception: Child Task 304073137 failed: Error building image: Error building purrr (0.2.2). Build exited with non-zero status: 1
>Execution halted

I'm fairly confused - `purrr` is installed and appears to be working on my machine. Google search tells me that this error occurs if the package is not updated to the CRAN version, but the version I have is the latest (0.2.2).

**Comment by philstraforelli** [comment]
To provide an update - I was able to publish using another computer. I'm still concerned that there's an issue with R Studio or at least some of the packages in my laptop though. Does anyone have any recommendation on what I should do to troubleshoot?

**Comment by emmagraham** [comment]
You could try clearing your environment and re-downloading all necessary packages. When you do this, I would also check to make sure the functionalities of purrr are not being masked by other packages. I would be great if the TAs could weigh in on this though, because I'm not sure.

**Comment by jennybc** [comment]
I haven't deployed to shinyapps.io in a long while, so I'm just giving general advice.

Are you running dev version of purrr perhaps? Maybe CRAN really means CRAN, no older and no newer?

Did you do whatever the equivalent of power cycling is? Probably de-installing and re-installing any packages that are giving you trouble. Also cleaning out the rsconnect or whatever directory gets created once you start using RStudio to deploy to remote places.

**Comment by samhinshaw** [comment]
First google result for "Error building purrr (0.2.2)" is...
https://groups.google.com/forum/#!msg/shinyapps-users/71yNwL8zdCs/NNzptfgBBQAJ

purrr's current version on CRAN is 0.2.2, so that doesn't seem the likely suspect, could it indeed by dplyr?

**Comment by jennybc** [comment]
Yes I think it does sound like a delicate issue of purrr vs dplyr versions. Hopefully you can wade through that thread and get to the same happy place as Andrew did. He's part of our local R community (hi @pssguy!), so he'd probably chat with you here. 😄

**Comment by philstraforelli** [comment]
I basically had to remove all of Hadley's packages using `remove.packages()` and re-installed them, and it's been fine since. Re-publishing the app worked on my main laptop.

I think the issue might have been in relation to an attempt to using the developmental version of `ggvis` on Hadley's github repo (I was hoping to use the facet option that is available in the package's developmental version but not in the CRAN version). I was suddenly getting a few errors during the installation phase, and it went downhill from there. Either way, I think things are back in order now. Thanks!

**Comment by pssguy** [comment]
@philstraforelli Good to hear. FYI I have had no such problems since mid October when I raised the issue on googlegroups. 
 Not sure if it is related at all but I do need to use deployApp() rather than the RStudio button when deploying to shinyapps.io
